File: sol/3217/3217.py
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        targets = frozenset(nums)
        foo = ListNode(-1, head)
        prev = foo
        ptr = head
        while ptr is not None:
            if ptr.val in targets:
                ptr = ptr.next
                prev.next = ptr
            else:
                prev = ptr
                ptr = ptr.next
        return foo.next

File: sol/3217/3217.cpp
#include <unordered_set>

struct Solution {
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int> targets(nums.begin(), nums.end());
        ListNode foo(-1, head);
        ListNode *prev = &foo, *ptr = head;
        while (ptr != nullptr){
            if (targets.contains(ptr->val)){
                ptr = ptr->next;
                prev->next = ptr;
            } else {
                prev = ptr;
                ptr = ptr->next;
            }
        }
        return foo.next;
    }
};
